src/js/api/posts/create.mjs

```diff
@@ -0,0 +1,30 @@
+// Functionality to create a post
+
+import { authFetch } from "../authFetch.mjs";
+import { apiSocialUrl } from "../constants.mjs";
+
+const action = "/posts";
+const method = "post";
+
+export async function createPost(postData) {
+    const createPostUrl = apiSocialUrl + action;
+
+    try {
+        const response = await authFetch(createPostUrl, {
+            method,
+            body: JSON.stringify(postData)
+        });
+        if (response.ok) {
+            location.reload();
+            return await response.json ();
+        } else {
+            throw new Error ("Response not ok");
+        }
+    } catch (error) {
+        throw new Error (error);
```

Comment:
The purpose of a catch block is to suppress an error.

The purpose of throw is to create an error.

There isn't really much point in catching an error, only to throw it again. I suggest you remove the throw from inside the catch block and consider catching this error somewhere else - in UI code, like a listener for example.